homeassistant/components/home_connect/diagnostics.py

```diff
@@ -0,0 +1,20 @@
+"""Diagnostics support for Home Connect Diagnostics."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntry
+
+from .const import (
+    DOMAIN,
+)
+
+async def async_get_config_entry_diagnostics(hass: HomeAssistant, entry: ConfigEntry) -> dict[str, Any]:
+    """Return diagnostics for a config entry."""
+    data = {}
+    for hc_api in hass.data[DOMAIN].values():
+        for device in hc_api.devices:
+            data[device.device_id] = device.appliance.status
```

Comment:
```suggestion
    for device in hass.data[DOMAIN][entry.entry_id].devices:
        data[device.device_id] = device.appliance.status
```
This method gathers a diagnostic from the given entry